controllers/authController.js

```diff
@@ -19,57 +19,69 @@ const signUp = async (req , res)  =>{
        res.status(422).json({error:"password must not be less than six characters"})
     }
     
-    await  Users.findOne({email : email})
-    .then(user =>{
-      if(user){
+const Email = await Users.findOne({email : email })
+
+    if (Email){
         res.status(422).json({error:"Email already registered"})
-      } else {
+      } 
 
-const newUser = new Users({name , email , username ,  password , password2})
+const User = await Users.findOne({username : username})
+
+    if(User){
+
+        res.status(422).json({error:"Username already exists!"})
 
-        //Hash password
 
-        bcrypt.genSalt(10 , ( err , salt )=>
+    }  else {
+const newUser = new Users({name , email , username ,  password , password2})
+        //Hash password
+        bcrypt.genSalt(10 , ( err , salt)=>
             bcrypt.hash(newUser.password, salt , async ( err , hash) =>{
-                if (err) throw err;
+                if(err) throw err;
                 newUser.password = hash
                 // save 
                 await newUser.save();
                 res.status(200).json("Registered");
             })
         )
 
+    }
 
-      }
-    })
+    }
+   
     
 
-}
+
 // Sign In
 
-const signIn = async  (req , res ) =>{
+const login = async  (req , res ) =>{
     const {username , password} = req.body;
+
     if(!username  || !password){
+
         res.json(422).json({error:'Incorrect Password!!!'})
```

Comment:
'Incorrect password' or 'All fields are required'?

Reply:
But you are checking if the frontend sent a `username` or `password`. Are you saying that if it is the `username` that is missing, the `password is incorrect`?

You are not checking for the correctness of the `password`, you are checking if they sent you the right fields